src/core/namespace.c

```diff
@@ -2791,21 +2790,17 @@ int bind_mount_add(BindMount **b, size_t *n, const BindMount *item) {
         if (!d)
                 return -ENOMEM;
 
-        c = reallocarray(*b, *n + 1, sizeof(BindMount));
-        if (!c)
+        if (!GREEDY_REALLOC_APPEND(*b, *n,
+                                   (&(BindMount) {
+                                           .source = TAKE_PTR(s),
+                                           .destination = TAKE_PTR(d),
+                                           .read_only = item->read_only,
+                                           .nosuid = item->nosuid,
+                                           .recursive = item->recursive,
+                                           .ignore_enoent = item->ignore_enoent,
+                                   }), 1))
```

Comment:
hmm, this would mean in case of OOM we invalidated s and d already, and thus would leak the two allocations